src/neuroconv/tools/testing/_mock/_mock_dataset_models.py

```diff
@@ -0,0 +1,57 @@
+from typing import Any, Dict, Iterable, Literal, Tuple, Union
+
+import h5py
+import numcodecs
+import numpy as np
+
+from ...nwb_helpers import (
+    AVAILABLE_HDF5_COMPRESSION_METHODS,
+    AVAILABLE_ZARR_COMPRESSION_METHODS,
+    DatasetInfo,
+    HDF5DatasetConfiguration,
+    ZarrDatasetConfiguration,
+)
+
+
+def mock_DatasetInfo() -> DatasetInfo:
```

Comment:
Why make this a function and not just a variable?

Reply:
Following the style of the PyNWB mocks: https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/src/pynwb/testing/mock/base.py

It definitely would not be a variable, I would appreciate a separate instance in memory each time

Probably not as relevant for the immutable one but very important for the mutable one